python/shotgun_fields/shotgun_field_manager.py

```diff
@@ -0,0 +1,244 @@
+        will be emitted.
+        """
+        if self._task_manager is None:
+            # create our own task manager if one wasn't passed in
+            task_manager = sgtk.platform.import_framework("tk-framework-shotgunutils", "task_manager")
+            self._task_manager = task_manager.BackgroundTaskManager(
+                parent=self,
+                max_threads=1,
+                start_processing=True
+            )
+
+        # let shotgun globals start loading the schema
+        shotgun_globals.register_bg_task_manager(self._task_manager)
+        shotgun_globals.run_on_schema_loaded(self.__schema_loaded)
+        self._initialized = True
+
+    def __schema_loaded(self):
+        """
+        Internal method that will be called when the schema is available.
+        """
+        self.initialized.emit()
+
+    def supported_fields(self, sg_entity_type, field_names):
+        """
+        Returns the subset of fields from field_names that have an associated widget class.
+
+        :param sg_entity_type: Shotgun entity type
+        :type sg_entity_type: String
+
+        :param field_names: An list of Shotgun field names
+        :type field_names: List of Strings
+
+        :returns: The subset of field_names that have associated widget classes.
+        """
+        supported_fields = []
+
+        # build a list of all the fields whose type is in the field map
+        for field_name in field_names:
+            data_type = shotgun_globals.get_data_type(sg_entity_type, field_name)
+            if data_type in self.__FIELD_TYPE_MAP:
+                supported_fields.append(field_name)
+
+        return supported_fields
+
+    def create_display_widget(self, sg_entity_type, field_name, entity=None, parent=None, **kwargs):
+        """
+        Returns the widget class associated with the field type if it has been registered.
+
+        If the entity is passed in and has the value for the requested field in it then the
+        initial contents of the widget will display that value.
+
+        Any keyword args other than those below will be passed to the constructor of whatever
+        QWidget the field widget wraps.
+
+        :param sg_entity_type: Shotgun entity type
+        :type sg_entity_type: String
+
+        :param field_name: Shotgun field name
+        :type field_name: String
+
+        :param entity: The Shotgun entity dictionary to pull the field value from.
+        :type entity: Whatever is returned by the Shotgun API for this field
+
+        :param parent: Parent widget
+        :type parent: :class:`PySide.QtGui.QWidget`
+
+        :returns: :class:`~PySide.QtGui.QWidget` or None if the field type has no display widget
+        """
+        data_type = shotgun_globals.get_data_type(sg_entity_type, field_name)
+
+        if data_type in self.__FIELD_TYPE_MAP:
+            cls = self.__FIELD_TYPE_MAP[data_type]
+
+            # instantiate the widget
+            return cls(
```

Comment:
The creation of the widget object and the setting of the value (via the `entity`) is now all handled in one go. Wondering if it might be cleaner to have to separate calls, one to construct the widget and then set the value? So basically first construct the widget, then call `obj.set_value()`? 

Without trying to guess too much, I suspect future additions to the `ShotgunFieldManager` may have the construction of widgets and the setting of values more decoupled (e.g. for delegates or keeping a pool of widgets around that are being reused or something).